src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -82,8 +89,7 @@ private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node,
             ExpressionSyntax patternExpression,
             bool hasErrors,
             DiagnosticBag diagnostics,
-            out bool wasExpression,
```

Comment:
Is this bool really saving any complexity? Can the caller just check if the converted expression has an error type?

Reply:
That is the implementation of this result, not its semantics. I expect it may be refined in the future. This is used to suppress a category of cascaded errors when the pattern expression bound to something other than a value.

---
In reply to: [154579877](https://github.com/dotnet/roslyn/pull/23209#discussion_r154579877) [](ancestors = 154579877)